packages/runed/src/lib/utilities/MediaQuery/MediaQuery.svelte.ts

```diff
@@ -34,43 +35,38 @@ import { browser } from "$lib/internal/utils/browser.js";
  * ```
  */
 export class MediaQuery {
-	#matches: boolean | undefined = $state();
+	#propQuery: MaybeGetter<string>;
+	#query = $derived.by(() => extract(this.#propQuery));
```

Comment:
Why `$derived.by` instead of `$derived(extract(this.propQuery))`? To avoid the TS error?

Reply:
Yeah, the alternative is to define `propQuery` like this
```ts
#propQuery!: MaybeGetter<string>;
```
This tells TS that the variable will be defined by the time it's accessed.